src/classes/UTIL_IntegrationConfig.cls

```diff
@@ -0,0 +1,147 @@
+        get {
+            if (namespace == null) {
+                namespace = initNamespace();
+            }
+            return namespace;
+        }
+        set;
+    }
+
+    @TestVisible
+    protected String callableApiClassName;
+
+    @TestVisible
+    protected System.Callable callableApi {
+        get {
+            if (callableApi == null) {
+                callableApi = initCallableApi();
+            }
+            return callableApi;
+        }
+        set;
+    }
+
+    /**********************************************************************************************************
+    * @description Checks if the package is installed
+    */
+    protected abstract Boolean initIsInstalled();
+
+    /**********************************************************************************************************
+    * @description Determines the package namespace
+    */
+    protected abstract String initNamespace();
+
+    /**********************************************************************************************************
+    * @description Confirms if the (un)managed package is installed
+    */
+    public Boolean isInstalled() {
+        return isInstalled;
+    }
+
+    /**********************************************************************************************************
+    * @description Instantiate the Callable API endpoint in the package
+    */
+    public System.Callable getCallableApi() {
+        return callableApi;
+    }
+
+    /**********************************************************************************************************
+    * @description Instantiate the Callable API endpoint in the package
+    */
+    private System.Callable initCallableApi() {
+        if (!isInstalled) {
+            return null;
+        }
+
+        return (Callable) Type.forName(namespace, callableApiClassName).newInstance();
+    }
+
+    /**********************************************************************************************************
+    * @description Adds the namespace before an SObject name or a child relationship name
+    */
+    public String prefixNamespace(String entity) {
+        if (String.isBlank(namespace)) {
+            return entity;
+        }
+
+        return namespace + '__' + entity;
+    }
+
+    public class DsoNpspPackage extends UTIL_IntegrationConfig {
+        /**********************************************************************************************************
+        * @description Constructor
+        */
+        public DsoNpspPackage() {
+            callableApiClassName = 'BridgeCallable_API';
+        }
+
+        /**********************************************************************************************************
+        * @description Checks if the package is installed
+        */
+        protected override Boolean initIsInstalled() {
+            return System.Type.forName(prefixNamespace('ArchiveRollupSummary__b')) != null;
```

Comment:
Where this kind of falls flat is if the customer decides to create their own `ArchiveRollupSummary__b` - then this would return true if the bridge was not installed. Is there a better way to check if an managed app is installed?

Reply:
You can look at UserInfo. isCurrentUserLicensed( string namespace) - that will tell you if the namespace is there.  It will throw an exception if the namespace doesn't exist. For unmanaged setups, you can't have two objects called ArchiveRollupSummary__b anyway.

Reply:
Can we support unmanaged external packages as well? Other package solutions have Custom Settings to support both managed and unmanaged packages as well as change of the namespace. Is it overhead to use the same approach here or there is a more elegant way to support (un)managed packages?

Reply:
@seanpat09 They way that was addressed in DSO was to check for two objects rather than just one, figuring that the chances of a customer creating two custom objects of their own with the exact same name of the objects in our package are really slim. Alternatively you could check for a field in an object (using describe).